storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -485,6 +485,8 @@ bool get_mysql_vars(MYSQL *connection) {
     have_backup_locks = true;
   }
 
+  if (opt_no_backup_locks) opt_lock_ddl = false;
```

Comment:
Why not validate this at xtrabackup.cc `if (opt_no_lock || opt_safe_slave_backup) opt_lock_ddl = false;` ?